File: Cpp/1310_XOR_Queries_of_a_Subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        vector<int> result(queries.size());
        vector<int> pre(arr.size());
        pre[0] = arr[0];
        for(int i=1; i<arr.size(); ++i){
            pre[i] = pre[i-1] ^ arr[i];
        }
        
        for(int i=0; i<queries.size(); ++i){
            if(queries[i][0] == 0) {
                result[i] = pre[queries[i][1]];
            }else {
                result[i] = pre[queries[i][0]-1] ^ pre[queries[i][1]];
            }
        }
        
        return result;
    }
};

File: Kotlin/1310_XOR_Queries_of_a_Subarray.kt
class Solution {
    fun xorQueries(arr: IntArray, queries: Array<IntArray>): IntArray {
        val result = IntArray(queries.size)
        val dp = IntArray(arr.count() + 1)
        
        for(i in 0 until arr.count()){
            dp[i + 1] = dp[i].xor(arr[i])
        }
        
        for(i in 0 until queries.count()){
            result[i] = dp[queries[i][1] + 1].xor(dp[queries[i][0]])

        }
        
        return result
    }
}